code/script.js

```diff
@@ -225,101 +232,163 @@ const generateBoard = () => {
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `Good thing this person isn't a ${value}. Remove all people that is.`
+      )
     }
-  }
+  } 
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  // Determine what is the category & filter by category to keep or remove based on the keep variable.
 
-    for accessories and other
+  if (category === 'hair' || category === 'eyes' ){
+    if (keep) {
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.category] === currentQuestion.value)
+    }
+    else {
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.category] !== currentQuestion.value)
+    }
+  }
+
+  if ( category === 'other' || category === 'accessories'){
+    if(keep) {
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    }
+    else {
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    }  
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
+  generateBoard() // Invoke a function to redraw the board with the remaining people.
 }
 
+start(); // Invokes the start function when website is loaded
+
+
+
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const confirmGuess = confirm(`You sure you want to guess on ${personToConfirm}?`)
+  if (confirmGuess) {
+    checkMyGuess(personToConfirm) //invoking the checkMyGuess function
+  }
 }
 
+
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+const checkMyGuess = (personToConfirm) => {
+  if (personToConfirm === secret.name) {
+   winOrLoseText1.innerHTML = `🥇 Hurray! 🥇`
+   winOrLoseText2.innerHTML = `You guessed the correct person, ${secret.name}!`
+  } else { winOrLoseText1.innerHTML = `Oh, bummer 🤷‍♀️ ...`
+            winOrLoseText2.innerHTML = `I'm sorry, but the correct person is ${secret.name}`
+  }
+
+
+winOrLose.style.display = "flex"
+board.style.display = "none"
 }
 
-// Invokes the start function when website is loaded
-start()
+
+  const playAgain = () => {
```

Comment:
You can remove one indentation here cuz' the playAgain function is not inside of any other curly brackets.